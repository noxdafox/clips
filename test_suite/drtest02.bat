(str-index "" "")                 ; DR0103 - 1
(str-index "" "a")                ; DR0103 - 2
(str-explode "")                  ; DR0104 - ()
(sub-string 1 2 "a")              ; DR0105 - "a"
(sub-string 2 2 "a")              ; DR0105 - ""
(subseq$ (create$ a) 2 2)     ; DR0106 - ()
(subseq$ (create$ a) 1 2)     ; DR0106 - (a)
(str-index "a" "aba")             ; DR0109 - 1
(str-implode)                     ; DR0112 - Error
(sub-string 1 3 "abc")            ; DR0113 - memory loss
(sub-string 1 3 "abc")            ; DR0113 - memory loss
(setgen 1)                        ; DR0114 - Error
(setgen a)                        ; DR0114 - Error
(read 1 1)                        ; DR0114 - Error
(readline 1 1)                    ; DR0114 - Error
(min)                             ; DR0114 - Error
(max)                             ; DR0114 - Error
(pi 1)                            ; DR0114 - Error
(halt 1)                          ; DR0114 - Error
(mod x y)                         ; DR0114 - Error
(setgen 1)                        ; DR0114 - Error
(gensym 10)                       ; DR0114 - Error
(if (eq 1 2) then)                ; DR0115 - Syntax
(if (eq 1 2) then else)           ; DR0115 - Syntax
(while (eq 1 2) do)               ; DR0115 - Syntax
(clear)                           ; DR0120
(defrule a 
   (x ?string) 
   => 
   (bind ?m (str-explode ?string)) 
   (bind ?str (str-implode ?m))
   (printout t ?m " " ?str crlf))
(defrule b 
   => 
   (bind ?m (str-explode "a b c 1 2 3"))
   (bind ?s (str-implode ?m))
   (printout t ?m " " ?s crlf))
(reset)                           ; DR0120
(assert (x "a b c 1 2 3"))        ; DR0120
(run)                             ; DR0120
(batch "a*a")                     ; DR0127
(open "a*a" a)                    ; DR0127
(fetch "a*a")                     ; DR0127
(clear)                           ; DR0135
(defrule rule1 ""
  (token ?level ?token-num)
  (expert ?level|* ?token-num|*)
  =>)
(assert (token 2 4))              ; DR0135
(assert (expert 2 4))             ; DR0135
(assert (expert 2 2))             ; DR0135
(assert (expert 2 *))             ; DR0135
(assert (expert * 4))             ; DR0135
(assert (expert * *))             ; DR0135
(assert (expert 4 *))             ; DR0135
(assert (expert * 2))             ; DR0135
(agenda)                          ; DR0135
(setgen 1)                        ; DR0156
(setgen -1)                       ; DR0156
(gensym)                          ; DR0156
(evenp dlaj)                      ; DR0159
(evenp 2e1)                       ; DR0159
(oddp 4.3)                        ; DR0159
(oddp 5e1)                        ; DR0159
(/ d 6)                           ; DR0160
(clear)                           ; DR0198
(deffacts bugfacts
   (base-count a 2 strand_13)
   (count-bases strand_13 [ t 3 ]))                
(defrule total
   ?accum <- (count-bases ?id $?bases-with-counts)
   (not (increment-base ? ?id ?))
   ?abase <- (base-count ?base ?count ?id)
   =>
   (retract ?accum ?abase)
   (bind ?bases-and-counts 
      (create$ [ ?base ?count ] ?bases-with-counts))
   (assert (count-bases ?id ?bases-and-counts)))
(reset)                           ; DR0198
(run)                             ; DR0198
(facts)                           ; DR0198
