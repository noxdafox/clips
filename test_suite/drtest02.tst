(unwatch all)
(clear)
(set-strategy depth)
(open "Results//drtest02.rsl" drtest02 "w")
(dribble-on "Actual//drtest02.out")
(batch "drtest02.bat")
(dribble-off)
(load "compline.clp")
(printout drtest02 "drtest02.bat differences are as follows:" crlf)
(compare-files "Expected//drtest02.out" "Actual//drtest02.out" drtest02)
; close result file
(close drtest02)
