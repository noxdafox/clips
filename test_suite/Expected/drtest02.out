TRUE
CLIPS> (batch "drtest02.bat")
TRUE
CLIPS> (str-index "" "")                 ; DR0103 - 1
1
CLIPS> (str-index "" "a")                ; DR0103 - 2
2
CLIPS> (str-explode "")                  ; DR0104 - ()
()
CLIPS> (sub-string 1 2 "a")              ; DR0105 - "a"
"a"
CLIPS> (sub-string 2 2 "a")              ; DR0105 - ""
""
CLIPS> (subseq$ (create$ a) 2 2)     ; DR0106 - ()
()
CLIPS> (subseq$ (create$ a) 1 2)     ; DR0106 - (a)
(a)
CLIPS> (str-index "a" "aba")             ; DR0109 - 1
1
CLIPS> (str-implode)                     ; DR0112 - Error
[ARGACCES4] Function str-implode expected exactly 1 argument(s)
CLIPS> (sub-string 1 3 "abc")            ; DR0113 - memory loss
"abc"
CLIPS> (sub-string 1 3 "abc")            ; DR0113 - memory loss
"abc"
CLIPS> (setgen 1)                        ; DR0114 - Error
1
CLIPS> (setgen a)                        ; DR0114 - Error
[ARGACCES5] Function setgen expected argument #1 to be of type integer
CLIPS> (read 1 1)                        ; DR0114 - Error
[ARGACCES4] Function read expected no more than 1 argument(s)
CLIPS> (readline 1 1)                    ; DR0114 - Error
[ARGACCES4] Function readline expected no more than 1 argument(s)
CLIPS> (min)                             ; DR0114 - Error
[ARGACCES4] Function min expected at least 2 argument(s)
CLIPS> (max)                             ; DR0114 - Error
[ARGACCES4] Function max expected at least 2 argument(s)
CLIPS> (pi 1)                            ; DR0114 - Error
[ARGACCES4] Function pi expected exactly 0 argument(s)
CLIPS> (halt 1)                          ; DR0114 - Error
[ARGACCES4] Function halt expected exactly 0 argument(s)
CLIPS> (mod x y)                         ; DR0114 - Error
[ARGACCES5] Function mod expected argument #1 to be of type integer or float
CLIPS> (setgen 1)                        ; DR0114 - Error
1
CLIPS> (gensym 10)                       ; DR0114 - Error
[ARGACCES4] Function gensym expected exactly 0 argument(s)
CLIPS> (if (eq 1 2) then)                ; DR0115 - Syntax
FALSE
CLIPS> (if (eq 1 2) then else)           ; DR0115 - Syntax
FALSE
CLIPS> (while (eq 1 2) do)               ; DR0115 - Syntax
FALSE
CLIPS> (clear)                           ; DR0120
CLIPS> (defrule a 
   (x ?string) 
   => 
   (bind ?m (str-explode ?string)) 
   (bind ?str (str-implode ?m))
   (printout t ?m " " ?str crlf))
CLIPS> (defrule b 
   => 
   (bind ?m (str-explode "a b c 1 2 3"))
   (bind ?s (str-implode ?m))
   (printout t ?m " " ?s crlf))
CLIPS> (reset)                           ; DR0120
CLIPS> (assert (x "a b c 1 2 3"))        ; DR0120
<Fact-1>
CLIPS> (run)                             ; DR0120
(a b c 1 2 3) a b c 1 2 3
(a b c 1 2 3) a b c 1 2 3
CLIPS> (batch "a*a")                     ; DR0127
[ARGACCES2] Function batch was unable to open file a*a.
FALSE
CLIPS> (open "a*a" a)                    ; DR0127
FALSE
CLIPS> (fetch "a*a")                     ; DR0127
[TEXTPRO1] Could not open file "a*a".
FALSE
CLIPS> (clear)                           ; DR0135
CLIPS> (defrule rule1 ""
  (token ?level ?token-num)
  (expert ?level|* ?token-num|*)
  =>)
CLIPS> (assert (token 2 4))              ; DR0135
<Fact-1>
CLIPS> (assert (expert 2 4))             ; DR0135
<Fact-2>
CLIPS> (assert (expert 2 2))             ; DR0135
<Fact-3>
CLIPS> (assert (expert 2 *))             ; DR0135
<Fact-4>
CLIPS> (assert (expert * 4))             ; DR0135
<Fact-5>
CLIPS> (assert (expert * *))             ; DR0135
<Fact-6>
CLIPS> (assert (expert 4 *))             ; DR0135
<Fact-7>
CLIPS> (assert (expert * 2))             ; DR0135
<Fact-8>
CLIPS> (agenda)                          ; DR0135
0      rule1: f-1,f-6
0      rule1: f-1,f-5
0      rule1: f-1,f-4
0      rule1: f-1,f-2
For a total of 4 activations.
CLIPS> (setgen 1)                        ; DR0156
1
CLIPS> (setgen -1)                       ; DR0156
[ARGACCES5] Function setgen expected argument #1 to be of type number (greater than or equal to 1)
1
CLIPS> (gensym)                          ; DR0156
gen1
CLIPS> (evenp dlaj)                      ; DR0159
[ARGACCES5] Function evenp expected argument #1 to be of type integer
CLIPS> (evenp 2e1)                       ; DR0159
[ARGACCES5] Function evenp expected argument #1 to be of type integer
CLIPS> (oddp 4.3)                        ; DR0159
[ARGACCES5] Function oddp expected argument #1 to be of type integer
CLIPS> (oddp 5e1)                        ; DR0159
[ARGACCES5] Function oddp expected argument #1 to be of type integer
CLIPS> (/ d 6)                           ; DR0160
[ARGACCES5] Function / expected argument #1 to be of type integer or float
CLIPS> (clear)                           ; DR0198
CLIPS> (deffacts bugfacts
   (base-count a 2 strand_13)
   (count-bases strand_13 [ t 3 ]))                
CLIPS> (defrule total
   ?accum <- (count-bases ?id $?bases-with-counts)
   (not (increment-base ? ?id ?))
   ?abase <- (base-count ?base ?count ?id)
   =>
   (retract ?accum ?abase)
   (bind ?bases-and-counts 
      (create$ [ ?base ?count ] ?bases-with-counts))
   (assert (count-bases ?id ?bases-and-counts)))
CLIPS> (reset)                           ; DR0198
CLIPS> (run)                             ; DR0198
CLIPS> (facts)                           ; DR0198
f-0     (initial-fact)
f-3     (count-bases strand_13 [ a 2 ] [ t 3 ])
For a total of 2 facts.
CLIPS> (dribble-off)
